### ISSUE TITLE
memory issues for convai2 seq2seq baseline

### ISSUE THREAD

**Issue by fgvbrt** [facebookresearch/ParlAI#843]
Could you please help me what are memory requirements for seq2seq baseline for convai2 competition:
https://github.com/facebookresearch/ParlAI/blob/master/projects/convai2/baselines/seq2seq/train.py
I got the following situation:
1) Batch size 64 and sgd optimizer crashes after first  evaluation
2) Batch size 32 and sgd optimizer crashes after few hours of work
3) Batch size 64 and adam optimizer crashes before first evaluation
4) Batch size 32 and adam in first few minutes.

I am using ParlAI commit aadd9886816b579e34bbeb3f2a85ffe4fb2eca5b
Pytorch 0.4.0,
Titan X 12gb
cuda-9.0, cudnn with cudnn v7.1

Is it normal behavior? It is very frustrating, that after few hours of training training everything could just crash. 

Example of crash below:
[crash.txt](https://github.com/facebookresearch/ParlAI/files/2091781/crash.txt)

Thanks in advance!

**Comment by alexholdenmiller** [comment]
The batch sizes there work for a 16gb GPU, but I'm not sure exactly what's pushing it over the edge after a few hours of training, as every example would have have been seen by then. It's possible it's getting ahead of pytorch's garbage collector. I pushed the GPU memory to the limit in order to not truncate the input sizes at all.

I'll dig around and see if I can find any places where memory usage might spike, but otherwise I've managed the memory usage just by setting the truncate parameter (e.g. `-tr 40`) or lowering the batchsize (however note that this can cut the persona from the history on this dataset).
